### ISSUE TITLE
unable to correctly compile with make 

### ISSUE THREAD

**Issue by huanfa** [bcgsc/arcs#63]
Hi
I have tried to compile the arcs but the Errors happened when using the make command.
Boost have installed correctly. Do you have any suggestions? Thanks in advance. 
```make  all-recursive
make[1]: Entering directory '/home/genetalks_gli_1/software/arcs-1.0.5'
Making all in Common
make[2]: Entering directory '/home/genetalks_gli_1/software/arcs-1.0.5/Common'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/genetalks_gli_1/software/arcs-1.0.5/Common'
Making all in DataLayer
make[2]: Entering directory '/home/genetalks_gli_1/software/arcs-1.0.5/DataLayer'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/genetalks_gli_1/software/arcs-1.0.5/DataLayer'
Making all in Graph
make[2]: Entering directory '/home/genetalks_gli_1/software/arcs-1.0.5/Graph'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/genetalks_gli_1/software/arcs-1.0.5/Graph'
Making all in Arcs
make[2]: Entering directory '/home/genetalks_gli_1/software/arcs-1.0.5/Arcs'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/genetalks_gli_1/software/arcs-1.0.5/Arcs'
Making all in Test
make[2]: Entering directory '/home/genetalks_gli_1/software/arcs-1.0.5/Test'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/genetalks_gli_1/software/arcs-1.0.5/Test'
make[2]: Entering directory '/home/genetalks_gli_1/software/arcs-1.0.5'
make[2]: Leaving directory '/home/genetalks_gli_1/software/arcs-1.0.5'
make[1]: Leaving directory '/home/genetalks_gli_1/software/arcs-1.0.5'
```

**Comment by lcoombe** [comment]
Hi @huanfa,
I don't see any errors there actually? It looks like you have already compiled ARCS in the past, and are re-compiling, but there are no new changes to any of the source files, so the Makefile sees that it doesn't have to do any recompilation.
Can you see the `arcs` executable in the `Arcs` directory?

**Comment by huanfa** [comment]
Thanks for your warmly reply. I have solved installing problem, but new issue had happened as follow:

- => Reading alignment files... Mon May 27 22:01:29 2019

- error: unexpected sequence: len=38000775 of size 38000775(base)
1. bam mapping
$bwa index -a bwtsw prefix.ctg.lay.2nd.fa 
${bwa} mem -t ${Threads} -M -R '@RG\tID:${i}\tLB:${i}\tPL:ILLUMINA\tSM:${i}' prefix.ctg.lay.2nd.fa \
/home/gonghuanfa/data/10x/R1.fastq \
/home/gonghuanfa/data/10x/R2.fastq > ${i}.sam
samtools view -@ ${Threads} -bS ${i}.sam > ${i}.bam
samtools sort -@ ${Threads} ${i}.bam -o ${i}.sort.bam
rm -rf ${i}.sam
samtools index -@ ${Threads} ${i}.sort.bam
2. arcks 
I have used the default parameter with arcks：
arcks -f prefix.ctg.lay.2nd.fa  -a bamlist -b result

Thanks in advance!

**Comment by lcoombe** [comment]
Hi @huanfa,
I suggest using our `arcs` Makefile (`Examples/arcs-make`) to run the pipeline, as this will do the alignment/arcs/scaffolding steps for you. We find a number of common errors that users make can be avoided by using that Makefile, or at least looking carefully at the structures of the steps in the Makefile.
In particular, it looks like you are not sorting the alignment file by read name? This is necessary for ARCS.

**Comment by huanfa** [comment]
@lcoombe I have sorted the contig name in bam file, but I noticed my _original.gv file was empty and all process information is "Warning: Skipped 1000000 unpaired reads......"

- - sh ~/software/arcs/Examples/arcs_test-demo/ARCS.sh  5 0.05 30000 draft.renamed.fa

1. 
I wondering the reason why I missing all the reads is that barcode information have not been used. Here are the 10x genomics raw data format:
BDHX190006596-1a-AK2003_L1_1.adapter.list.gz  BDHX190006596-2a-AK1647_L1_1.adapter.list.gz
BDHX190006596-1a-AK2003_L1_1.fq.gz                BDHX190006596-2a-AK1647_L1_1.fq.gz
BDHX190006596-1a-AK2003_L1_2.adapter.list.gz  BDHX190006596-2a-AK1647_L1_2.adapter.list.gz
BDHX190006596-1a-AK2003_L1_2.fq.gz                 BDHX190006596-2a-AK1647_L1_2.fq.gz

and I only use the renamed fq.gz file to generate a barcode.fastq.gz based on the longranger basic

2.barcode.fastq.gz file format, no BX:Z tag in the file
@A00129:382:HJW5VDSXX:1:1546:19859:29309
TACCCTCATTTGAGGCATAAAACATAGTAGAAAAGGCTGGCAGTGAACTTTCTGCCCCTACAAATGTAAGAGTATCATCCCCGTGGGGTTTGGGAACAGACCGCGCTGGGTAAGCATCAGGCTTATG
+
:FFFFF:FFFF,:FFFFFFFFFFF:F,FF,FFFFFFFFFFFFFFFFF:FFF,FFFFFFFFFFF,F:FFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFF,FFFFFFF:FFFFFFFFF,FFF:

Is there any problem on the getting a barcoded fastq file from the 10x genomics raw data? Thanks.
